platforms/core-configuration/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -119,8 +122,9 @@ abstract class GeneratePrecompiledScriptPluginAccessors @Inject internal constru
 
     @get:InputFiles
     @get:Classpath
-    val runtimeClassPathFiles: FileCollection
-        get() = runtimeClassPathArtifactCollection.get().artifactFiles
+    @get:ReplacesEagerProperty
+    val runtimeClassPathFiles: Provider<FileCollection>
```

Comment:
This feels a bit weird, but it passes the tests :/

 Not sure if there is a better way to handle that

Reply:
Yeah, this can stay just `FileCollection`.

Note: These plugins are internal so we don't need `@get:ReplacesEagerProperty` and we can just merge them directly to master

Reply:
But FileCollection isn't provider-friendly, is it?

Reply:
It is. If you have properties without setter, then we use FileCollection, otherwise we use ConfigurableFileCollection

Reply:
@asodja I've rebased to master and removed `ReplacesEagerProperty`, PTAL